client/my-sites/patterns/pages/home/index.tsx

```diff
@@ -51,16 +50,11 @@ export const PatternsHomePage = ( {
 				title="Build your perfect site with patterns"
 			/>
 
-			<PatternsGridGallery
+			<CategoryGallery
 				title="Ship faster with patterns"
 				description="Choose from a huge library of patterns to build any page you need."
-				list={ categories?.map( ( category ) => ( {
-					name: category.name,
-					label: category.label,
-					number: category.regularPatternCount,
-					image: ImgPattern,
-					link: `/patterns/${ category.name }`,
-				} ) ) }
+				categories={ categories?.filter( ( c ) => c.regularPatternCount ) }
```

Comment:
As for me, `regularPatternCount` and `pagePatternCount` is not universal flag. WDUYT about changing the structure returned from backend to:

```
type: 'pattern' | 'layout'
count: number;
```

Reply:
Also it will simplify rendering things like this https://github.com/Automattic/wp-calypso/pull/88367/files#diff-903e9a11c071084a8bd558363a8287076d24c47407d1a45e3fad310e032fabc9R78-R80

Reply:
Just my two cents: I would avoid adding the `type: layout` for pages because we are moving out from that "layout" concept. We still have v1 patterns tagged as "layout", but in v2, they are just big patterns in the category `page`.

These page patterns are also used as starter patterns. Read more at https://developer.wordpress.org/news/2024/01/31/adding-starter-patterns-to-your-wordpress-themes/

Reply:
It's the main thing which concerns me - namings. We provide these two types of patterns as different types, since they are separated in UI. So if we separate them - it would be good to have own namings, at least internally (inside A8C). to have specific and understandable namings. Also it will simplify communication for us as developers, we will be more clear.
And this part of code is good example - when I am reviewing it, I feel a bit lost and sometimes I am not sure - which exactly React-component I am looking at.

Reply:
Categories can contain both regular patterns and page patterns, so currently, `regularPatternCount` and `pagePatternCount` are not mutually exclusive. Maybe that was part of the confusion here..?

Reply:
@fredrikekelund  Yep, now I see